common/core/hwclayerbuffermanager.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+#include "hwclayerbuffermanager.h"
+
+namespace hwcomposer {
+
+HwcLayerBufferManager::HwcLayerBufferManager() {
+  for(size_t i = 0; i < BUFFER_CACHE_LENGTH; i++)
+    cached_buffers_.emplace_back();
+}
+
+HwcLayerBufferManager::~HwcLayerBufferManager() {
+    PurgeBuffer();
+}
+
+void HwcLayerBufferManager::PurgeBuffer() {
+  for(auto &map : cached_buffers_) {
```

Comment:
This should be called in CompositorThread only. I think CompositorThread owing the ownership of texture, eglimage of buffer makes sense. We need to make sure that context is current in this thread before releasing these resources.

Reply:
Will make it private

Reply:
This is where we release gl resources https://github.com/intel/IA-Hardware-Composer/blob/master/common/compositor/compositorthread.cpp#L116  we need to make sure all this is sync. Only way is to move destruction of buffers to compositor thread.

Reply:
You mean we should call purge in here?https://github.com/intel/IA-Hardware-Composer/blob/master/common/compositor/compositorthread.cpp#L116 ?
It makes sense

Reply:
ya

Reply:
We can't do it as ReleaseGpuResources is called each Draw frame, and it will make cache for GL resources not meaningful?

Reply:
I think what we want to do is after every frame post  a task to clean up any buffers which can be destroyed. So, this PurgeBuffer should be called by posting a task to CompositorThread from DisplayQueue after you have updated commit for the current frame. This task should only posted when previous frame had involved gpu composition otherwise you could call PergeBuffer from DisplayQueue itself